src/main/webapp/src/app/ranking/ranking-item/ranking-item.component.ts

```diff
@@ -23,8 +23,11 @@ export class RankingItemComponent implements OnInit{
     }
 
     // This prevent the tests from failing with "Cannot read property '0' of undefined"
-    if (this.hiddenGem.types != null && this.hiddenGem.types[0] != null) {
-      this.type = this.hiddenGem.types[0];
+    if (this.hiddenGem.types != null) {
+      if (this.hiddenGem.types[0] != null && (this.hiddenGem.types[0] === 'restaurant' || this.hiddenGem.types[0] === 'cafe'))
```

Comment:
Should this be a loop along the length of the type array? or does something guarantee there are only 2?

Reply:
There are usually more than 2 types in the array, but how I filter the hidden gems in the backend is that they take into consideration businesses which have types 'restaurant' and 'cafe' as 1st type or 2nd type. 
In the frontend, I also want to display only 'restaurant' or 'cafe' as type (and not some other types like 'bar', 'meal_takeway', etc.)

Reply:
ok in that case could you put a comment explaining that the backend guarantees the types we car about will be in the first two elements.

Reply:
Done